storagenode/contact/chore.go

```diff
@@ -48,60 +58,107 @@ func (chore *Chore) Run(ctx context.Context) (err error) {
 	defer mon.Task()(&ctx)(&err)
 	chore.log.Info("Storagenode contact chore starting up")
 
+	var group errgroup.Group
+
 	if !chore.service.initialized.Wait(ctx) {
 		return ctx.Err()
 	}
 
-	return chore.Loop.Run(ctx, func(ctx context.Context) error {
-		if err := chore.pingSatellites(ctx); err != nil {
-			chore.log.Error("pingSatellites failed", zap.Error(err))
-		}
-		return nil
-	})
+	backOff := time.Second
+	interval := chore.interval
+	chore.mu.Lock()
+	for _, satellite := range chore.trust.GetSatellites(ctx) {
+		satellite := satellite
+
+		cycle := sync2.NewCycle(interval)
+		chore.cycles = append(chore.cycles, cycle)
+
+		cycle.Start(ctx, &group, func(ctx context.Context) error {
+			chore.log.Info("starting contact cycle for satellite " + satellite.String())
+			interval := backOff
+			attempts := 0
+			for {
+				err := chore.pingSatellite(ctx, satellite)
+				attempts++
+				if err == nil {
+					return nil
+				}
+				chore.log.Error("ping satellite failed " + strconv.Itoa(attempts) + " times: " + err.Error())
+
+				if !sync2.Sleep(ctx, interval) && interval == chore.interval {
```

Comment:
@egonelbre is this correct here?

Reply:
I don't think it's necessary.

If you want to break out when you hit the standard interval, then you can just as well return in `if interval > chore.interval {` block. Adding the check here makes it more confusing.

Reply:
That's true for that case. Will the loop continue after 1 second the first time it reaches this block?

Reply:
nevermind, it's working as intended

Reply:
nevermind, I had a misunderstanding of how sync2 sleep was working.

Reply:
It should be good to go now. Thoroughly tested.

Reply:
I wanted to break out of the loop at the standard interval, particularly to make it clear that the main cycle is still running at a regular interval.